src/imageloader.js

```diff
@@ -120,6 +120,9 @@ $.ImageJob.prototype = {
      * @memberof OpenSeadragon.ImageJob#
      */
     finish: function(data, request, dataType) {
+        if (!this.jobId) {
```

Comment:
I am not sure I want to have this, but to fix existing demos this should be here. It prevents the application to call `finish` handler when `abort` was called beforehand. The demo tested `0` timeout, there were warnings in the console, but the app rendered the viewport without problems anyway. This  'fixes' the issue, meaning if the timeout happens, the data will not get delivered. On the other hand, being able to recover is also a feature...

Reply:
Well, of course we can update the demos. I guess the question is whether we expect that behavior in the wild.

And if we want to be able to recover, would that be an extra API or something that happens automatically?

Reply:
I am not sure. It's just that although the call has timed out, the server was able to somehow deliver the data in the end, and finish succsefully. There is for now no extra api. There could be potential issues (meanwhile the tile has been loaded again succesfully and now the previous job has also finished, or the tile is no longer needed and the data just initiates pointless computation). But if it finishes a short time after timeout it likely resurrects the tile :D It seems like it happens automatically.

Note that I went through these old demos and fixed most (if not all) of them.

Reply:
One scenario, I'm not sure if it's related: If the user closes the tiledImage and opens a new one, we want to be able to cancel all of the requests for the old image and have them not come through. I know in the past we've had reports from people seeing tiles from closed images show up and throw warnings into the console.